contracts/utils/convex/ConvexYieldWrapper.sol

```diff
@@ -69,17 +82,22 @@ contract ConvexStakingWrapperYield is ConvexStakingWrapper {
             return 0;
         }
 
-        if (vaults[account_].length == 0) {
-            return _balanceOf[account_];
-        }
         bytes12[] memory userVault = vaults[account_];
 
         //add up all balances of all vaults
         uint256 collateral;
         DataTypes.Balances memory balance;
         for (uint256 i = 0; i < userVault.length; i++) {
-            balance = cauldron.balances(userVault[i]);
-            collateral = collateral + balance.ink;
+            if (userVault[i] != bytes12(0)) {
+                if (cauldron.vaults(userVault[i]).owner == account_) {
+                    balance = cauldron.balances(userVault[i]);
+                    collateral = collateral + balance.ink;
+                }
+                // else {
```

Comment:
I think that adding this call would be fine.

Reply:
Adding this would lead to `_getDepositedBalance` no longer being a view only function.